### PR TITLE
Add resource definitions for docker repositories

### PR DESCRIPTION
**Pull request by TuningYourCode**

### CODE REVIEW

File: manifests/resource/repository/docker/proxy.pp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+# @param docker_proxy_index_url
+#   If docker_proxy_index_type is CUSTOM you have to set the uri of the index api.
+#
+# @example
+#   nexus::repository::docker::proxy { 'docker-docker.io':
+#      proxy_remote_url => 'https://registry-1.docker.io',
+#   }
+#
+define nexus::resource::repository::docker::proxy (
+  Stdlib::HTTPSUrl $proxy_remote_url,
```
Comment (func0der) [inline manifests/resource/repository/docker/proxy.pp:47]: Should we default this to the default docker registry?
I mean, are the so many alternatives?

File: manifests/resource/repository/docker/proxy.pp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        'enabled'    => $negative_cache_enabled,
+        'timeToLive' => $negative_cache_time_to_live
+      },
+      'httpClient'      => {
+        'blocked'        => $http_client_blocked,
+        'autoBlock'      => $http_client_auto_block,
+        'connection'     => {
+          'retries'                 => undef,
+          'userAgentSuffix'         => undef,
+          'timeout'                 => undef,
```
Comment (func0der) [inline manifests/resource/repository/docker/proxy.pp:95]: Why are we defining `undef` here? Are we overwriting low defaults?
Are those recommended defaults by sonatype?

**Review by func0der — APPROVED** [review]

File: manifests/resource/repository/docker/proxy.pp (hunk 1 above)
Comment (TuningYourCode) [reply manifests/resource/repository/docker/proxy.pp:47]: https://codefresh.io/docs/docs/integrations/docker-registries/

Podman also asks (i think on the first usage) which registry should be used as default. Not sure about docker but i guess it defaults to docker hub.

File: manifests/resource/repository/docker/proxy.pp (hunk 2 above)
Comment (TuningYourCode) [reply manifests/resource/repository/docker/proxy.pp:95]: We didn't implement / use these settings yet (also not on apt/npm proxy) but if we do not have the keys the api response will differ from the definition and puppet would do api calls to change the resource on every run - which we obviously don't want.

**Pull request merged**